adserver/tasks.py

```diff
@@ -218,6 +232,128 @@ def daily_update_keywords(day=None):
                 )
 
 
+@app.task()
+def daily_update_regiontopic(day=None):  # pylint: disable=too-many-branches
+    """
+    Update the RegionTopicImpression index each day.
+
+    Each data point will have one region, but multiple possible topics.
+
+    :arg day: An optional datetime object representing a day
+    """
+    start_date, end_date = _get_day(day)
+
+    log.info("Updating RegionTopic's for %s-%s", start_date, end_date)
+
+    # Remove all old impressions, because they are cumulative
+    RegionTopicImpression.objects.filter(
+        date__gte=start_date,
+        date__lt=end_date,
+    ).delete()
+
+    for impression_type in ["views"]:
+        queryset = _default_filters(impression_type, start_date, end_date)
+
+        unsold = {}
+        no_keywords = {}
+
+        for values in (
+            queryset.values("publisher", "advertisement", "keywords", "country")
+            .annotate(total=Count("country"))
+            .filter(total__gt=0)
+            .order_by("-total")
+            .values(
+                "publisher",
+                "advertisement",
+                "keywords",
+                "country",
+                "advertisement__flight__targeting_parameters",
+                "total",
+            )
+        ):
+            if not (
+                values["keywords"]
+                and values["country"]
+                and values["advertisement__flight__targeting_parameters"]
```

Comment:
Are there a lot of flights that don't have targeting?

Reply:
Not sure. Probably mostly house ads.